java_tests/src/test/java/org/lsst/sal/SAL_TestTest.java

```diff
@@ -0,0 +1,427 @@
+        remote.salEventSub("Test_logevent_arrays");
+        sleep(STD_SLEEP);
+        controller.salEventPub("Test_logevent_arrays");
+        sleep(STD_SLEEP);
+        boolean[] getEvents = {false, true};
+
+        for (boolean getEvent : getEvents) {
+            getNewestAfterGetOldest(true, getEvent);
+        }
+    }
+
+    /**
+     * Test that get newest after get oldest gets the newest value.
+     * <p>
+     * This tests DM-18491.
+     *
+     * @throws Exception In case of a problem related to reflection.
+     */
+    @Test
+    public void testTelGetNewestAfterGetOldest() throws Exception {
+        remote.salTelemetrySub("Test_arrays");
+        sleep(STD_SLEEP);
+        controller.salTelemetryPub("Test_arrays");
+        sleep(STD_SLEEP);
+
+        getNewestAfterGetOldest(false, false);
+    }
+
+    /**
+     * Test that get newest after get oldest gets the newest value.
+     * <p>
+     * Uses the arrays topic. This tests DM-18491.
+     *
+     * @param testEvents If True then test events else test telemetry.
+     * @param getEvent   If True then use getEvent to get the oldest value, else use getNextSample_logevent_scalars.
+     *                   Ignored if test_events is False.
+     * @throws Exception In case of a problem related to reflection.
+     */
+    private void getNewestAfterGetOldest(boolean testEvents, boolean getEvent) throws Exception {
+        final int numLoops = 5;
+        Object[] dataList = new Object[numLoops];
+        Object data;
+        if (testEvents) {
+            data = new logevent_arrays();
+        } else {
+            data = new arrays();
+        }
+
+        int retcode;
+        for (int i = 0; i < numLoops; i++) {
+            TestUtils.fillArraysWithRandomValues(data);
+            dataList[i] = data;
+            if (testEvents) {
+                retcode = controller.logEvent_arrays((logevent_arrays) data, 1);
+            } else {
+                retcode = controller.putSample((arrays) data);
+            }
+            sleep(STD_SLEEP);
+            Assert.assertEquals(retcode, SAL_Test.SAL__OK);
+        }
+
+        // read and check the oldest value
+        Object expected_data = dataList[0];
+        if (testEvents) {
+            if (getEvent) {
+                retcode = remote.getEvent_arrays((logevent_arrays) data);
+            } else {
+                retcode = remote.getNextSample((logevent_arrays) data);
+            }
+        } else {
+            retcode = remote.getNextSample((arrays) data);
+        }
+        sleep(STD_SLEEP);
+        Assert.assertEquals(retcode, SAL_Test.SAL__OK);
+        TestUtils.assertArraysEqual(expected_data, data);
+
+        // read and check the newest value
+        expected_data = dataList[numLoops - 1];
+        getTopic("getSample", data);
+        TestUtils.assertArraysEqual(expected_data, data);
+    }
+
+    /**
+     * Test that a late joiner can an event using getNextSample.
```

Comment:
"can an event" -> "can read the most recent event" or similar.

Reply:
Again, I implemented the python tests as literal as possible meaning that I copied the python doc strings as well. I've fixed this in both the python and the Java files.